### ISSUE TITLE
Email not sent when entering address into 'Email link to person' field

### ISSUE THREAD

**Issue by 0scillator** [nextcloud/server#2442]
### Steps to reproduce
1. click 'share link'
2. both: enter mail address into 'Email link to person'-field manually OR via autocomplete
3. hit enter

### Expected behaviour
should send email notification to entered address

My email configuration works. I normally can send emails and I can do it via 'Test email'

### Actual behaviour
no email is sent
logfile reads: 'Error'   'php'   'Undefined index: id at /var/www/nextcloud/core/ajax/share.php#230'

### Server configuration
**Operating system**:
Ubuntu-server 14.04.5 LTS
**Web server:**
apache2
**Database:**
mysql  Ver 14.14 Distrib 5.5.53, for debian-linux-gnu (x86_64) using readline 6.3
**PHP version:**
PHP 5.5.9-1ubuntu4.20 (cli) (built: Oct  3 2016 13:00:37)
**Nextcloud version:** (see Nextcloud admin page)
10.0.1
**Updated from an older Nextcloud/ownCloud or fresh install:**
fresh install, data and dbs copied from a former owncloud installation months ago
**Where did you install Nextcloud from:**
official download
https://download.nextcloud.com/server/releases/nextcloud-10.0.1.zip
**Signing status:**
<details>
<summary>Signing status</summary>
No errors have been found.
```
Login as admin user into your Nextcloud and access 
http://example.com/index.php/settings/integrity/failed 
paste the results here.
```
</details>

**List of activated apps:**
<details>
<summary>App list</summary>
Enabled:
  - activity: 2.3.2
  - calendar: 1.4.1
  - comments: 1.0.0
  - contacts: 1.4.0.0
  - dav: 1.0.1
  - federatedfilesharing: 1.0.1
  - federation: 1.0.1
  - files: 1.5.2
  - files_external: 1.0.2
  - files_pdfviewer: 0.8.1
  - files_sharing: 1.0.0
  - files_texteditor: 2.1
  - files_trashbin: 1.0.0
  - files_versions: 1.3.0
  - files_videoplayer: 0.9.8
  - gallery: 15.0.0
  - notifications: 0.3.0
  - password_policy: 1.0.0
  - provisioning_api: 1.0.0
  - serverinfo: 1.1.1
  - survey_client: 0.1.5
  - systemtags: 1.0.2
  - tasks: 0.9.3
  - templateeditor: 0.1
  - theming: 1.0.1
  - updatenotification: 1.0.1
  - workflowengine: 1.0.1
Disabled:
  - admin_audit
  - encryption
  - external
  - files_accesscontrol
  - files_automatedtagging
  - files_retention
  - firstrunwizard
  - user_external
  - user_ldap
  - user_saml
```
If you have access to your command line run e.g.:
sudo -u www-data php occ app:list
from within your Nextcloud installation folder
```
</details>

**The content of config/config.php:**
<details>
<summary>Config report</summary>
{
    "system": {
        "instanceid": "occ793a7b891",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "*****",
            "*****",
            "*****",
            "*****"
        ],
        "datadirectory": "\/var\/www\/owncloud\/data",
        "dbtype": "mysql",
        "version": "9.1.1.5",
        "dbname": "owncloud",
        "dbhost": "localhost",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "forcessl": true,
        "maxZipInputSize": 1073741824,
        "allowZipDownload": true,
        "theme": "",
        "maintenance": false,
        "mail_smtpmode": "smtp",
        "mail_from_address": "nextcloud",
        "mail_domain": "fuchsfuchs.at",
        "mail_smtphost": "localhost",
        "mail_smtpport": "25",
        "overwritewebroot": "\/nextcloud",
        "overwrite.cli.url": "\/nextcloud",
        "loglevel": 3,
        "enable_previews": true,
        "preview_max_scale_factor": 10,
        "preview_libreoffice_path": "\/usr\/bin\/libreoffice",
        "preview_office_cl_parameters": "",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "memcache.local": "\\OC\\Memcache\\APCu",
        "trashbin_retention_obligation": "auto",
        "updatechecker": false,
        "htaccess.RewriteBase": "\/nextcloud"
    },
    "apps": {
        "activity": {
            "enabled": "yes",
            "installed_version": "2.3.2",
            "types": "filesystem"
        },
        "admin_audit": {
            "enabled": "no",
            "installed_version": "1.0.0",
            "types": "logging"
        },
        "backgroundjob": {
            "lastjob": "2"
        },
        "bookmarks": {
            "enabled": "no",
            "installed_version": "0.7",
            "ocsid": "168710",
            "types": ""
        },
        "calendar": {
            "enabled": "yes",
            "installed_version": "1.4.1",
            "ocsid": "168707",
            "types": ""
        },
        "comments": {
            "enabled": "yes",
            "installed_version": "1.0.0",
            "types": "logging"
        },
        "contacts": {
            "enabled": "yes",
            "installed_version": "1.4.0.0",
            "ocsid": "168708",
            "types": ""
        },
        "core": {
            "backgroundjobs_mode": "cron",
            "global_cache_gc_lastrun": "1430999101",
            "installedat": "1404252558.431",
            "lastcron": "1480606202",
            "lastupdateResult": "[]",
            "lastupdatedat": "1480606480",
            "oc.integritycheck.checker": "[]",
            "public_caldav": "calendar\/share.php",
            "public_calendar": "calendar\/share.php",
            "public_documents": "documents\/public.php",
            "public_files": "files_sharing\/public.php",
            "public_gallery": "gallery\/public.php",
            "public_webdav": "dav\/appinfo\/v1\/publicwebdav.php",
            "remote_caldav": "dav\/appinfo\/v1\/caldav.php",
            "remote_calendar": "dav\/appinfo\/v1\/caldav.php",
            "remote_carddav": "dav\/appinfo\/v1\/carddav.php",
            "remote_contacts": "dav\/appinfo\/v1\/carddav.php",
            "remote_core.css": "\/core\/minimizer.php",
            "remote_core.js": "\/core\/minimizer.php",
            "remote_dav": "dav\/appinfo\/v2\/remote.php",
            "remote_files": "dav\/appinfo\/v1\/webdav.php",
            "remote_filesync": "files\/appinfo\/filesync.php",
            "remote_mozilla_sync": "mozilla_sync\/appinfo\/remote.php",
            "remote_webdav": "dav\/appinfo\/v1\/webdav.php",
            "repairlegacystoragesdone": "yes",
            "shareapi_allow_links": "yes",
            "shareapi_allow_mail_notification": "yes",
            "shareapi_allow_public_notification": "yes",
            "shareapi_allow_public_upload": "yes",
            "shareapi_allow_resharing": "yes",
            "shareapi_default_expire_date": "no",
            "shareapi_enabled": "yes",
            "shareapi_enforce_links_password": "no",
            "shareapi_exclude_groups": "yes",
            "shareapi_exclude_groups_list": "[\"NOSHARE\"]",
            "shareapi_expire_after_n_days": "14",
            "shareapi_only_share_with_group_members": "yes",
            "shareapi_share_policy": "groups_only",
            "umgmt_show_backend": "false",
            "umgmt_show_email": "false",
            "umgmt_show_last_login": "true",
            "umgmt_show_storage_location": "false",
            "vendor": "nextcloud"
        },
        "dav": {
            "enabled": "yes",
            "installed_version": "1.0.1",
            "types": "filesystem"
        },
        "documents": {
            "converter": "local",
            "enabled": "no",
            "installed_version": "0.13.1",
            "ocsid": "168711",
            "types": ""
        },
        "federatedfilesharing": {
            "enabled": "yes",
            "installed_version": "1.0.1",
            "types": ""
        },
        "federation": {
            "enabled": "yes",
            "installed_version": "1.0.1",
            "types": "authentication"
        },
        "files": {
            "backgroundwatcher_previous_file": "323",
            "backgroundwatcher_previous_folder": "163",
            "cronjob_scan_files": "500",
            "default_quota": "10 GB",
            "enabled": "yes",
            "installed_version": "1.5.2",
            "types": "filesystem"
        },
        "files_external": {
            "\/Dropbox": "1417015134",
            "\/GoogleDrive": "1416163488",
            "allow_user_mounting": "yes",
            "enabled": "yes",
            "google::1049162735466-ak9n7299a0svqrtu1472724190cId": "46049",
            "installed_version": "1.0.2",
            "ocsid": "166048",
            "types": "filesystem",
            "user_mounting_backends": "dav,owncloud,sftp,dropbox,googledrive,\\OC\\Files\\Storage\\SFTP_Key"
        },
        "files_pdfviewer": {
            "enabled": "yes",
            "installed_version": "0.8.1",
            "ocsid": "166049",
            "types": ""
        },
        "files_sharing": {
            "enabled": "yes",
            "installed_version": "1.0.0",
            "outgoing_server2server_share_enabled": "no",
            "types": "filesystem"
        },
        "files_texteditor": {
            "enabled": "yes",
            "installed_version": "2.1",
            "ocsid": "166051",
            "types": ""
        },
        "files_trashbin": {
            "enabled": "yes",
            "installed_version": "1.0.0",
            "types": "filesystem"
        },
        "files_versions": {
            "enabled": "yes",
            "installed_version": "1.3.0",
            "types": "filesystem"
        },
        "files_videoplayer": {
            "enabled": "yes",
            "installed_version": "0.9.8",
            "types": ""
        },
        "files_videoviewer": {
            "enabled": "no",
            "installed_version": "0.1.3",
            "ocsid": "166054",
            "types": ""
        },
        "firstrunwizard": {
            "enabled": "no",
            "installed_version": "1.1",
            "ocsid": "166055",
            "types": ""
        },
        "gallery": {
            "enabled": "yes",
            "installed_version": "15.0.0",
            "types": ""
        },
        "mozilla_sync": {
            "enabled": "no",
            "installed_version": "1.4",
            "quota_limit": "0",
            "types": ""
        },
        "notifications": {
            "enabled": "yes",
            "installed_version": "0.3.0",
            "types": "logging"
        },
        "password_policy": {
            "enabled": "yes",
            "installed_version": "1.0.0",
            "types": ""
        },
        "provisioning_api": {
            "enabled": "yes",
            "installed_version": "1.0.0",
            "types": "prevent_group_restriction"
        },
        "search_lucene": {
            "enabled": "no",
            "installed_version": "0.6.0",
            "ocsid": "168709",
            "types": "filesystem"
        },
        "serverinfo": {
            "enabled": "yes",
            "installed_version": "1.1.1",
            "types": ""
        },
        "survey_client": {
            "enabled": "yes",
            "installed_version": "0.1.5",
            "types": ""
        },
        "systemtags": {
            "enabled": "yes",
            "installed_version": "1.0.2",
            "types": "logging"
        },
        "tasks": {
            "enabled": "yes",
            "installed_version": "0.9.3",
            "ocsid": "164356",
            "types": ""
        },
        "templateeditor": {
            "enabled": "yes",
            "installed_version": "0.1",
            "types": ""
        },
        "theming": {
            "enabled": "yes",
            "installed_version": "1.0.1",
            "types": "logging"
        },
        "updatenotification": {
            "enabled": "yes",
            "installed_version": "1.0.1",
            "types": ""
        },
        "updater": {
            "enabled": "no",
            "installed_version": "0.4",
            "ocsid": "166059",
            "types": ""
        },
        "user_webdavauth": {
            "enabled": "no",
            "installed_version": "1.1.0.0",
            "types": "authentication"
        },
        "workflowengine": {
            "enabled": "yes",
            "installed_version": "1.0.1",
            "types": "filesystem"
        }
    }
}
```
If you have access to your command line run e.g.:
sudo -u www-data php occ config:list system
from within your Nextcloud installation folder

or 

Insert your config.php content here
(Without the database password, passwordsalt and secret)
```
</details>

**Are you using external storage, if yes which one:** WebDAV

**Are you using encryption:** yes

**Are you using an external user-backend, if yes which one:** no

### Client configuration
**Browser:**
Tested with several browsers...
**Operating system:**
... on several operating systems, but right now KDE-Neon 5.8.4
### Logs
as above

#### Web server error log
<details>
<summary>Web server error log</summary>
```
Insert your webserver log here
```
</details>

#### Nextcloud log (data/nextcloud.log)
<details>
<summary>Nextcloud log</summary>
{"reqId":"ddXWkBtf5WLx3c9\/596h","remoteAddr":"***.**.*.**","app":"PHP","message":"Undefined index: id at \/var\/www\/nextcloud\/core\/ajax\/share.php#230","level":3,"time":"2016-12-01T14:51:57+00:00","method":"GET","url":"\/nextcloud\/index.php\/core\/ajax\/share.php?fetch=getShareWithEmail&search=alexander%40**********","user":"alexander"}
```
Insert your Nextcloud log here
```
</details>

#### Browser log
<details>
<summary>Browser log</summary>
As mentioned above, I tried with several browsers on several operating systems
```
Insert your browser log here, this could for example include:

a) The javascript console log
b) The network log
c) ...
```
</details>

**Comment by nickvergessen** [comment]
Tested while fixing https://github.com/nextcloud/server/issues/1764

and works all fine for me. So this might be fixed already in the upcoming 10.0.2 release.

**Comment by 0scillator** [comment]
Solved with 10.0.2
